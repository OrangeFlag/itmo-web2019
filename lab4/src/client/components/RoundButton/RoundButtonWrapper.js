import styled from 'styled-components';

const RoundButtonWrapper = styled.button`
    border: 0;
    padding: 13px 16px;
    // height: 50px;
    // line-height: 50px;
    // border-radius: ___CSS_0___;
    // box-shadow: ___CSS_1___;
    outline: none;
    display: inline-block;
    text-align: center;
    text-decoration: none;
    font-size: 15px;
    border-radius: 4px;

    &:hover {
        text-decoration: none;
        background: #cccccc;
        color: ___CSS_3___;
    }
   

    &:focus {
        text-decoration: none;
        outline: none;
    }
`;

export default RoundButtonWrapper