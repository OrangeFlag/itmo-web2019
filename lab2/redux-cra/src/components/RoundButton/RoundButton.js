import React from 'react';
import PropTypes from "prop-types";
import RoundButtonWrapper from "./RoundButtonWrapper";

export const RoundButton = (props) => {
    return (
        <RoundButtonWrapper onClick={props.action} type={props.type || "button"}>
            {props.children}
        </RoundButtonWrapper>

    );
};

RoundButton.propTypes = {
    action: PropTypes.func,
    type: PropTypes.string
};