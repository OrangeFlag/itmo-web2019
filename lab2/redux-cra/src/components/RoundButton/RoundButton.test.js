import React from 'react';
import {mount} from 'enzyme';
import {RoundButton} from './RoundButton';

describe('Button tests', () => {
    it('should render correctly +', () => {
        const component = mount(<RoundButton>+</RoundButton>);

        expect(component).toMatchSnapshot();
        component.unmount();
    });

    it('should render correctly with type', () => {
        const component = mount(<RoundButton type={"reset"}>+</RoundButton>);

        expect(component).toMatchSnapshot();
        component.unmount();
    });

    it('button click should call function', () => {
        const clickFn = jest.fn();

        const component = mount(<RoundButton action={clickFn}>-</RoundButton>);

        component.simulate('click');

        expect(clickFn).toHaveBeenCalled();
        component.unmount();
    });
});