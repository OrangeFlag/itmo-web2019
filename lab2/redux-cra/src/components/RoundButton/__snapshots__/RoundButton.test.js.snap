// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should render correctly + 1`] = `
<RoundButton>
  <styled.button
    type="button"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "JEQYc",
            "rules": Array [
              "
    border: 0;
    padding: 13px 16px;
    // height: 50px;
    // line-height: 50px;
    // border-radius: ___CSS_0___;
    // box-shadow: ___CSS_1___;
    outline: none;
    display: inline-block;
    text-align: center;
    text-decoration: none;
    font-size: 15px;
    border-radius: 4px;

    &:hover {
        text-decoration: none;
        background: #cccccc;
        color: ___CSS_3___;
    }
   

    &:focus {
        text-decoration: none;
        outline: none;
    }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      type="button"
    >
      <button
        className="sc-bdVaJa JEQYc"
        type="button"
      >
        +
      </button>
    </StyledComponent>
  </styled.button>
</RoundButton>
`;

exports[`Button tests should render correctly with type 1`] = `
<RoundButton
  type="reset"
>
  <styled.button
    type="reset"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "JEQYc",
            "rules": Array [
              "
    border: 0;
    padding: 13px 16px;
    // height: 50px;
    // line-height: 50px;
    // border-radius: ___CSS_0___;
    // box-shadow: ___CSS_1___;
    outline: none;
    display: inline-block;
    text-align: center;
    text-decoration: none;
    font-size: 15px;
    border-radius: 4px;

    &:hover {
        text-decoration: none;
        background: #cccccc;
        color: ___CSS_3___;
    }
   

    &:focus {
        text-decoration: none;
        outline: none;
    }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      type="reset"
    >
      <button
        className="sc-bdVaJa JEQYc"
        type="reset"
      >
        +
      </button>
    </StyledComponent>
  </styled.button>
</RoundButton>
`;
